rectools/visuals/visual_app.py

```diff
@@ -144,20 +195,153 @@ def _group_recos(
             prepared_recos[model_name] = prepared_model_recos
         return prepared_recos
 
-    def _check_columns_present_in_recos(self, recos: TablesDict) -> None:
-        required_columns = {self.id_col, Columns.Item}
+    @classmethod
+    def _ungroup_recos(
+        cls,
+        grouped_recos: tp.Dict[tp.Hashable, TablesDict],
+        selected_requests: tp.Dict[tp.Hashable, tp.Hashable],
+        id_col: str,
+    ) -> pd.DataFrame:
+        res = []
+        for model_name, prepared_model_recos in grouped_recos.items():
+            for request_name, request_recos in prepared_model_recos.items():
+                df = request_recos.copy()
+                df[id_col] = selected_requests[request_name]
+                df[Columns.Model] = model_name
+                res.append(df)
+        return pd.concat(res, axis=0)
+
+    @classmethod
+    def _ungroup_interactions(
+        cls,
+        grouped_interactions: TablesDict,
+        selected_requests: tp.Dict[tp.Hashable, tp.Hashable],
+        id_col: str,
+    ) -> pd.DataFrame:
+        res = []
+        for request_name, request_interactions in grouped_interactions.items():
+            df = request_interactions.copy()
+            df[id_col] = selected_requests[request_name]
+            res.append(df)
+        return pd.concat(res, axis=0)
```

Comment:
Probably it makes sense to reset index here

Reply:
Sorry. Fixed now. Thanks)